src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -512,7 +538,22 @@ private async Task<(bool, string)> GetManifestPackageUpdate(SdkFeatureBand sdkFe
             }
         }
 
-        private string GetAdvertisingManifestPath(SdkFeatureBand featureBand, ManifestId manifestId) =>
+        private async Task<(bool, NuGetVersion)> GetPackageVersion(PackageId packageId, PackageSourceLocation packageSourceLocation = null, bool includePreview = false)
+        {
+            try
+            {
+                var latestVersion = await _nugetPackageDownloader.GetLatestPackageVersion(packageId, packageSourceLocation: _packageSourceLocation, includePreview: includePreview);
+                return (true, latestVersion);
```

Comment:
@joeloff Do we use this pattern of returning (Success + Value) frequently, I would rather return just null and check that instead of having a separate boolean to check. Might want to add some exception or warning docstring that this can return null?

Reply:
@nagilson  this is related to a change that Daniel and I made together when implementing fallback in a PR from before you started. we are continuing that same pattern of behavior here.